PWGHF/D2H/TableProducer/candidateCreatorB0Reduced.cxx

```diff
@@ -44,19 +46,23 @@ struct HfCandidateCreatorB0Reduced {
   Configurable<double> minParamChange{"minParamChange", 1.e-3, "stop iterations if largest change of any B0 is smaller than this"};
   Configurable<double> minRelChi2Change{"minRelChi2Change", 0.9, "stop iterations is chi2/chi2old > this"};
   // selection
-  Configurable<double> invMassWindowB0{"invMassWindowB0", 0.3, "invariant-mass window for B0 candidates"};
+  Configurable<double> invMassWindowDPiTolerance{"invMassWindowDPiTolerance", 0.01, "invariant-mass window tolerance for DPi pair preselections (GeV/c2)"};
+  // variable that will store the value of invMassWindowDPi (defined in dataCreatorDplusPiReduced.cxx)
+  float myInvMassWindowDPi{1.};
 
-  double massPi = RecoDecay::getMassPDG(kPiPlus);
-  double massD = RecoDecay::getMassPDG(pdg::Code::kDMinus);
-  double massB0 = RecoDecay::getMassPDG(pdg::Code::kB0);
-  double massDPi{0.};
+  // O2DatabasePDG service
+  Service<o2::framework::O2DatabasePDG> pdg;
+
+  double massPi{0.};
+  double massD{0.};
+  double massB0{0.};
   double bz{0.};
 
   // Fitter for B vertex (2-prong vertex filter)
   o2::vertexing::DCAFitterN<2> df2;
 
-  Preslice<aod::HfCand3ProngReduced> candsDPerCollision = hf_track_index_reduced::hfReducedCollisionId;
-  Preslice<aod::HfTracksReduced> tracksPionPerCollision = hf_track_index_reduced::hfReducedCollisionId;
+  Preslice<soa::Join<aod::HfRed3Prongs, aod::HfRed3ProngsCov>> candsDPerCollision = hf_track_index_reduced::hfRedCollisionId;
```

Comment:
In the `candidateCreatorB0.cxx`` there D-pi candidates are filtered accounting for the response of the candidateSelectorD0 (see here: https://github.com/AliceO2Group/O2Physics/blob/c8f778a304ee9723d1d05a95dd47a240fded21e6/PWGHF/TableProducer/candidateCreatorB0.cxx#L90)
while while in your case you are not doing it. This is because you already do the D0 selections in dataCreatorDplusPiReduced, right?

Reply:
Yes, same answer as for Bplus.